### PR TITLE
Update repository and server interface for enabling git push

### PR DESCRIPTION
**Pull request by JeongHunP**

### CODE REVIEW

File: repository/tests/integration_test.rs [hunk 1]
```diff
@@ -140,3 +142,150 @@ async fn basic_1() {
...
+        .collect();
+    client_node_repo
+        .finalize(
+            block_commit,
+            FinalizationProof {
+                signatures,
+                round: 0,
+            },
+        )
+        //.sync(&block.to_hash256(), &block_proof)
```
Comment (junha1) [inline repository/tests/integration_test.rs:279]: Please delete this

File: repository/src/interpret/push.rs [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+    Ok(is_eligible)
+}
+
+pub async fn broadcast(
+    raw: &mut RawRepository,
+    private_key: Option<PrivateKey>,
+) -> Result<(), Error> {
+    let agendas = read_agendas(raw).await?;
+    let blocks = read_blocks(raw).await?;
+    let remotes = raw.list_remotes().await?;
```
Comment (junha1) [inline repository/src/interpret/push.rs:32]: Could you add `read_governance_approved_agendas()` and implement it? (currently it's todo, but I'm sure it's easy)

File: repository/src/interpret/push.rs [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+            branch.into(),
+            Some(format!(
+                "{commit_hash} {branch} {timestamp} {signature} {signer}"
+            )),
+        )
+        .await?;
+        raw.delete_branch(branch.into()).await?;
+    }
+    Ok(())
+}
```
Comment (junha1) [inline repository/src/interpret/push.rs:113]: I thinks this function has too many duplicated code. Please put the target branches in a same list and iterate over it

**Review by junha1 — COMMENTED** [review]
Please add new line at the end of each hook

File: repository/tests/integration_test.rs (hunk 1 above)
Comment (JeongHunP) [reply repository/tests/integration_test.rs]: fixed

File: repository/src/interpret/push.rs (hunk 2 above)
Comment (JeongHunP) [reply repository/src/interpret/push.rs]: Implemented.

File: repository/src/interpret/push.rs (hunk 3 above)
Comment (JeongHunP) [reply repository/src/interpret/push.rs:69]: fixed

**Review by junha1 — APPROVED** [review]

**Pull request merged**